pyiceberg/table/__init__.py

```diff
@@ -3410,6 +3411,94 @@ def _readable_metrics_struct(bound_type: PrimitiveType) -> pa.StructType:
             schema=entries_schema,
         )
 
+    def partitions(self) -> "pa.Table":
```

Comment:
TODO: Waiting on https://github.com/apache/iceberg-python/pull/599 to merge to introduce TimeTravel